core/src/types/storage.rs

```diff
@@ -1161,13 +1178,12 @@ pub struct Epochs {
     /// Invariant: the values must be sorted in ascending order.
     pub first_block_heights: Vec<BlockHeight>,
 }
-
 impl Default for Epochs {
-    /// Initialize predecessor epochs, assuming starting on the epoch 0 and
-    /// block height 0.
+    /// Initialize predecessor epochs, assuming start of the first epoch at
+    /// block height 1.
     fn default() -> Self {
         Self {
-            first_block_heights: vec![BlockHeight::default()],
+            first_block_heights: vec![BlockHeight(1)],
```

Comment:
this is still somewhat incorrect, though, no? because it assumes height 1 has already been committed even before we commit the first block

Reply:
yeah, good point! I think we can just initialize it empty as it's later set from init chain

Reply:
@tzemanovic if it doesn't break the protocol, sure, but I think it's still semantically wrong. when calling finalize-block for the first time, no block has been committed yet, so `Epochs` in theory would be empty

Reply:
I think it's ok - the first epoch is initialized on `InitChain` instead of `FinalizeBlock` as for many sub-systems it needs different logic than the epoch change in `FinalizeBlock`

Reply:
done in [6fb1984](https://github.com/anoma/namada/pull/1993/commits/6fb1984cbdc0522031c5e1c46e4c83b5b8f794d2)